os/fs/smartfs/smartfs_smart.c

```diff
@@ -277,10 +280,34 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 
 		if (parentdirsector != 0xFFFF) {
 			/* We can create in the given parent directory */
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+			if (smartfs_findinvalidentry(fs, parentdirsector, &invalidentry)) {
 
-			ret = smartfs_createentry(fs, parentdirsector, filename, SMARTFS_DIRENT_TYPE_FILE, mode, &sf->entry, 0xFFFF, sf);
-			if (ret != OK) {
-				goto errout_with_buffer;
+				invalidentry.firstsector = 0xFFFF;
```

Comment:
It always has invalid firstsector, because entry is already invalidated.

Reply:
Got it.